### DISCUSSION TITLE
Stuck at DungeonDB Start

### DISCUSSION THREAD

**Discussion by HeadcaseFargone** [mbround18/valheim-docker#180] [Help!]
Since running auto_update.sh I am able to connect to my server but it's a completely fresh world.  I noticed that the .db file in /worlds was now 42MB whereas in my backup from before the update it was 66MB.  I stopped the container, copied over the .db and .fwl (everything from /worlds) to the volume and restarted the container.  Now it never makes it past this line:

02/25/2021 08:34:08: DungeonDB Start 279

**Comment by lilasquared** [comment]
@HeadcaseFargone I just followed a few simple steps to update the game server.  In the readme he has env vars you can set to ensure the server auto updates.

https://github.com/mbround18/valheim-docker/discussions/125#discussioncomment-378682

**Reply by HeadcaseFargone** [reply]
Okay so despite the lack of a "Game Server connected" message I was eventually able to log into my old world intact.  I had to wait a good 10 minutes before it would respond though.  Very odd, but at least it's working now.

**Reply by mbround18** [reply]
Pull the `:latest` version and restart your server 👍

**Reply by Phenyx** [reply]
I'm kind of having the same issue, except after waiting roughly 30 minutes, the game hasn't loaded yet.  It does spin messages out about unloading unused assets and does world saves, but it won't allow connections. I've also made sure to pull an update as well.

Using the original Dedicated file, the game loads.

-OS fresh Ubuntu 20.04 installation.
-Game server save came from a windows 10 box. (I don't know if this has any relevance.)
  - All four files have been copied, both server_name.fwl + .old and server_name.db + .old
  - chown/chmod settings are cloned from the server files it spun up. (It reads the files fine it looks like.)

The logs last output:
```bash
02/25/2021 13:52:53: DungeonDB Start 761

(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

02/25/2021 14:02:53:  Connections 0 ZDOS:251046  sent:0 recv:0
```

If you have any suggestions, I'll run through them, thank you!

**Reply by Phenyx** [reply]
Figured out one issue!  Valheim does not appreciate passwords shorter than 6 characters...  
I'm currently seeing "Game server connected" but it won't let me connect yet.  I'll go brew some coffee and wait a bit.

**Reply by CosmicHorrorDev** [reply]
Hmmm, we currently have a check for shorter than 5 characters. I'll double check that you are right, but we might have to update it to 6 characters if this is the case

**Reply by mbround18** [reply]
lol yeah valheim changed the password req XP

**Reply by Phenyx** [reply]
I may try running this locally on the system and not within a container to see if it loads, but I'm going to wait 30 minutes.  Still unable to connect.

 The current log file is at:
```bash
(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

02/25/2021 14:37:44: DungeonDB Start 308

(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

02/25/2021 14:47:44:  Connections 0 ZDOS:251046  sent:0 recv:0

(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

Unloading 6 Unused Serialized files (Serialized files now loaded: 0)

Unloading 6 unused Assets to reduce memory usage. Loaded Objects now: 1116725.
Total: 479.639501 ms (FindLiveObjects: 65.717628 ms CreateObjectMapping: 16.965700 ms MarkObjects: 396.180663 ms  DeleteObjects: 0.774746 ms)
```

**Reply by Phenyx** [reply]
Also, I extremely appreciate the work that was done on this, even if my issue does not get solved! :)

**Reply by mbround18** [reply]
your server is loaded btw, those logs are correct for a started system. Open the game and try joining by IP

**Reply by mbround18** [reply]
@Phenyx deleted for safety <3 mind if i try connecting?

**Reply by Phenyx** [reply]
That's why I posted it.  Please do. <3

**Reply by mbround18** [reply]
Huh yeah I get disconnected when trying to connect (unless you shutdown, my uhh player name is Thiccy)

**Reply by Phenyx** [reply]
Yeah, didn't shut down.  It's still up.

**Reply by CosmicHorrorDev** [reply]
I also got disconnected trying a 5 character password. Testing out 6 now but my laptop is not appreciating all the image building

**Reply by Phenyx** [reply]
Here are the current logs:
```bash
root@ns501342:/home/boulvier/valheim/valheim/server/logs# tail -f ./*
==> ./auto-backup.out <==

==> ./auto-update.out <==

==> ./output.log <==

==> ./valheim_server.err <==
RecordSteamInterfaceCreation (PID 60): SteamUtils009 / Utils
Setting breakpad minidump AppID = 892970
RecordSteamInterfaceCreation (PID 60): SteamGameServer012 / GameServer
RecordSteamInterfaceCreation (PID 60): SteamUtils009 / Utils
RecordSteamInterfaceCreation (PID 60): SteamNetworking005 / Networking
RecordSteamInterfaceCreation (PID 60): SteamGameServerStats001 / GameServerStats
RecordSteamInterfaceCreation (PID 60): STEAMHTTP_INTERFACE_VERSION002 / HTTP
RecordSteamInterfaceCreation (PID 60): STEAMINVENTORY_INTERFACE_V002 / Inventory
RecordSteamInterfaceCreation (PID 60): STEAMUGC_INTERFACE_VERSION010 / UGC
RecordSteamInterfaceCreation (PID 60): STEAMAPPS_INTERFACE_VERSION008 / Apps

==> ./valheim_server.log <==
(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

02/25/2021 14:57:46: Saved 251046 zdos

(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

02/25/2021 14:57:46: World saved ( 812.399ms )

(Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

```

May I switch the file the docker-compose.yml over to Dedicated and see if we can still connect to validate I'm not doing something silly?

**Reply by mbround18** [reply]
on your host do you have the server directory mounted? If so can you drag and drop your full server log to here?

```
/home/steam/server/logs/valheim_server.log
```

**Reply by Phenyx** [reply]
absolutely, one second.

**Reply by mbround18** [reply]
@LovecraftianHorror tested with a 5 character password with no issue :P

**Reply by CosmicHorrorDev** [reply]
Ahhh, my local test server install was out of date and for some reason the usual old version check wasn't showing up in the logs v.v

An `odin install` later and 5 character password _is_ working as expected

@Phenyx is your installation out of date as well? If someone can dm me the ip and port I can check remotely. Or if your server files are mounted you can check the `buildid` in `server/steamapps/appmanifest_896660.acf`. It should be `6287700`

**Comment by Phenyx** [comment]
[valheim_server.log](https://github.com/mbround18/valheim-docker/files/6045765/valheim_server.log)

**Reply by Phenyx** [reply]
I'm in a putty shell so I had to cat the whole file. XD

**Reply by mbround18** [reply]
lol, thats a log of a healthy server

**Reply by mbround18** [reply]
Whats your network to this server look like? Is it hosted on a platform?

**Reply by Phenyx** [reply]
Yeah.  It's kind of a really weird one.  If I were to docker-compose down, edit the server name to something new or to the original Dedicated name, then docker-compose up, the server lets me connect and pops up a password prompt.

**Reply by mbround18** [reply]
What server name are you using?

**Reply by Phenyx** [reply]
It's an actual machine with a direct IP address.  I've not tossed additional IP addresses to it yet, though.  It's hosted via OVH in Canada.

**Reply by Phenyx** [reply]
Ashbringer (or Dedicated to test)

**Reply by Phenyx** [reply]
Permissions are correct, too.  Dedicated was the server file generated by the docker instance, which I did to ensure I had correct ownership/chmod perms when uploading the windows server files.
```bash
# ls -lha
total 68M
drwxr-x--- 2 ubuntu ubuntu 4.0K Feb 25 20:57 .
drwxr-xr-x 3 ubuntu ubuntu 4.0K Feb 25 06:16 ..
-rw-r----- 1 ubuntu ubuntu  29M Feb 25 20:57 Ashbringer.db
-rw-r----- 1 ubuntu ubuntu  29M Feb 25 20:37 Ashbringer.db.old
-rw-r----- 1 ubuntu ubuntu   46 Feb 25 20:57 Ashbringer.fwl
-rw-r----- 1 ubuntu ubuntu   46 Feb 25 20:37 Ashbringer.fwl.old
-rw-r----- 1 ubuntu ubuntu 5.2M Feb 25 19:06 Dedicated.db
-rw-r----- 1 ubuntu ubuntu 5.2M Feb 25 18:57 Dedicated.db.old
-rw-r----- 1 ubuntu ubuntu   45 Feb 25 19:06 Dedicated.fwl
-rw-r----- 1 ubuntu ubuntu   45 Feb 25 18:57 Dedicated.fwl.old
```

**Reply by mbround18** [reply]
Love OVH, thats where I have my dedicated server hosted as well. Would you mind dropping your docker-compose file here? Minus the password ofc.

**Reply by Phenyx** [reply]
```yaml
version: "3"
services:
  valheim:
    image: mbround18/valheim:latest
    ports:
      - 2456:2456/udp
      - 2457:2457/udp
      - 2458:2458/udp
    environment:
      - PORT=2456
      - NAME="Boulvier's DragonKnights"
      - WORLD="Ashbringer"
      - PASSWORD="REDACTATHON"
      - TZ=America/Chicago
      - PUBLIC=1
      - AUTO_UPDATE=0
      - DEBUG_MOD=0 #I did set this to 1 to see if I could get odin to give me more love.
    volumes:
    - ./valheim/saves:/home/steam/.config/unity3d/IronGate/Valheim
    - ./valheim/server:/home/steam/valheim
```

**Reply by mbround18** [reply]
LOL can you change it to DEBUG_MODE=1 (my typo plagues me XP ) and change the name to `Boulviers DragonKnights`

**Reply by Phenyx** [reply]
Absolutely.  So, here's an update.  I haven't changed any configs on either system or compose except for name.. but I last tested and verified dedicated worked, but then about 2 hours ago after the last update, I have pulled the latest container.  Last night was when I started this process.

I just tested Dedicated, that does not let me connect at all. Rebooting the server and I'll get you that debug_mode=1 output.

**Reply by Phenyx** [reply]
[valheim_server.log](https://github.com/mbround18/valheim-docker/files/6045844/valheim_server.log)

**Comment by Phenyx** [comment]


**Reply by CosmicHorrorDev** [reply]
I found out my issue was from an out of date install and the server logs no longer showing a version check as a heads up. I wonder if you're hitting the same issue.

**Reply by Phenyx** [reply]
Huh.  Where would I see that to confirm?  You're also having trouble getting the server to start, too, on latest?  It worked for me last night before the update earlier on Dedicated, so I just made the assumption it could very well be the windows generated world files...

**Reply by CosmicHorrorDev** [reply]
I have a local test image i use for debugging and developing that I was using to troubleshoot.

if your server files are mounted you can check the buildid in `server/steamapps/appmanifest_896660.acf`. It should be 6287700 if it's up to date.

And if you are out of date then you can force an update and restart the server by

```txt
$ docker exec -it --user steam $CONTAINER_NAME bash
steam@container $ odin stop
steam@container $ odin install
steam@container $ odin start
```

**Reply by mbround18** [reply]
lol could you nuke your server installation and `docker-compose down|up` ? to see if you are missing the latest patches

**Reply by Phenyx** [reply]
yep, hold a second.  I'm ending the world and everything we know it as it was.

**Reply by Phenyx** [reply]
TADA! You both gave me the right answer. :)

Damn, and I was hoping auto_update=1 would of done all the magic there. XD

**Reply by CosmicHorrorDev** [reply]
Woohoo! \o/

And hopefully it will be more magical soon (#177)

Well good to know that the server logs no longer indicate a version miss-match

**Reply by Phenyx** [reply]
Okok, should I build into my script an auto-update that will just nuke the steam server dir? XD

**Reply by CosmicHorrorDev** [reply]
Also just to clarify. The default auto update schedule currently only checks for an update once a day. So an update coming in at a bad time can still lead to using an old version for a while.

**Reply by Phenyx** [reply]
I mean, I'm loving this project.  I just hope he's not frustrated with us.

**Reply by CosmicHorrorDev** [reply]
I don't think he's been bugged by me _quite_ yet and people talking about their issues in the discussions help us know how to make the image more resilient :D

**Reply by mbround18** [reply]
Lol yall are all good :P and I am glad your servers up :)
